readthedocs/proxito/views/serve.py

```diff
@@ -322,11 +323,12 @@ def get(self, request, proxito_path, template_name='404.html'):
         # If that doesn't work, attempt to serve the 404 of the current version (version_slug)
         # Secondly, try to serve the 404 page for the default version
         # (project.get_default_version())
-        doc_type = (
+        version = (
             Version.objects.filter(project=final_project, slug=version_slug)
-            .values_list('documentation_type', flat=True)
+            .only("documentation_type")
             .first()
         )
+        doc_type = version.documentation_type if version else None
```

Comment:
Why did we change this logic? It seems the same to me?

Reply:
We need the version object now, before we were using just the documentation type (values_list).